ReactiveCocoa/Swift/Scheduler.swift

```diff
@@ -57,25 +61,23 @@ public final class ImmediateScheduler: SchedulerType {
 /// If the caller is already running on the main queue when an action is
 /// scheduled, it may be run synchronously. However, ordering between actions
 /// will always be preserved.
-public final class UIScheduler: SchedulerType {
-	private static var dispatchOnceToken: dispatch_once_t = 0
-	private static var dispatchSpecificKey: UInt8 = 0
+public final class UIScheduler: SchedulerProtocol {
+	private static var __once: () = {
+			DispatchQueue.main.setSpecific(key: UIScheduler.dispatchSpecificKey,
+				value: UIScheduler.dispatchSpecificContext)
+		}()
+	private static var dispatchOnceToken: Int = 0
+	private static var dispatchSpecificKey = DispatchSpecificKey<UInt8>()
 	private static var dispatchSpecificContext: UInt8 = 0
 
 	private var queueLength: Int32 = 0
 
 	public init() {
-		dispatch_once(&UIScheduler.dispatchOnceToken) {
-			dispatch_queue_set_specific(
-				dispatch_get_main_queue(),
-				&UIScheduler.dispatchSpecificKey,
-				&UIScheduler.dispatchSpecificContext,
-				nil
-			)
-		}
+		_ = UIScheduler.__once
```

Comment:
Can you add a comment explaining that this is to eagerly initialize the static property?